metabase/metabase
# Markdown support in the pinned items

Closes https://github.com/metabase/metabase/issues/30381

I found the following issues while working on this:
- #31326
- #31327
- #31328

### Description

Adds markdown support in tooltips in pinned items.
- Headings and links are disabled in tooltips (to be consistent with changes in #29890)


### How to verify

1. Create a:
    - Dashboard
    - Model
    - Question
2. Add description to it:
    - markdown with short / long text
    - plain text with short / long text
3. Pin it to a dashboard
4. If it's a question, turn the visualization on and off
5. In the pinned item hover the text description (or description icon) to display the tooltip - it should show markdown without headings and links formatted
6. If there is a description block in the card, it should contain only the first description line with markdown formatting removed

To verify skeletons, check out the description tooltip when card is loading. To keep it in that state forever you can block requests to `api/card/*/query` URL in your developer tools.

### Demo

#### Before

https://github.com/metabase/metabase/assets/6830683/bf7ba71f-3eed-4a42-8309-8765ad09915b


#### After

https://github.com/metabase/metabase/assets/6830683/d501b58e-32a9-4e1b-a4cb-1f642dda9b70
